### ISSUE TITLE
Enhance the DataFilter documentation

### ISSUE THREAD

**Issue by machour** [yiisoft/yii2#19173]
The current documentation is **frustrating**

The concept is introduced in the [REST quick start](https://www.yiiframework.com/doc/guide/2.0/en/rest-quick-start#trying-it-out) in a sentence. 

In that sentence, we are invited to read the [Resource](https://www.yiiframework.com/doc/guide/2.0/en/rest-resources#filtering-collections) section for more information, which just displays another sentence, inviting us to:

 * read the Quick Start for an example (I was just there)
 * the data provider section on [Data Filters](https://www.yiiframework.com/doc/guide/2.0/en/output-data-providers#filtering-data-providers-using-data-filters) which yields a small example for a single model filtering.

There are no information to be found on how to filter on a relation, how to properly form GET queries with advanced filters, ..

A real-world example would be more than welcome.

**Comment by machour** [comment]
Also, clearly talk about `ActiveDataFilter` and `DataFilter` in this page https://www.yiiframework.com/doc/guide/2.0/en/output-data-providers#filtering-data-providers-using-data-filters

The link at the bottom is responsible for wasting me 3 hours today.

**Comment by bizley** [comment]
Yes, totally agree. I also spent huge time on this when trying to grasp the idea and do some enhancements. The problem is as usual with no time available to write better docs...
